yt_dlp/extractor/soundcloud.py

```diff
@@ -147,14 +154,17 @@ def genNumBlock():
             'user_agent': self._USER_AGENT
         }
 
-        query = self._API_AUTH_QUERY_TEMPLATE % self._CLIENT_ID
-        login = sanitized_Request(self._API_AUTH_URL_PW % query, json.dumps(payload).encode('utf-8'))
-        response = self._download_json(login, None)
-        self._access_token = response.get('session').get('access_token')
-        if not self._access_token:
-            self.report_warning('Unable to get access token, login may has failed')
-        else:
-            self._HEADERS = {'Authorization': 'OAuth ' + self._access_token}
+        response = self._download_json(
+            self._API_AUTH_URL_PW % (self._API_AUTH_QUERY_TEMPLATE % self._CLIENT_ID),
+            None, note='Verifying login token...', fatal=False,
+            data=json.dumps(payload).encode())
+
+        if token := traverse_obj(response, ('session', 'access_token', {str})):
+            self._HEADERS = {'Authorization': f'OAuth {token}'}
+            self.report_login()
+            return
+
+        raise ExtractorError('Unable to get access token, login may have failed', expected=True)
```

Comment:
?

Reply:
this is user/pass login code that has been commented like this for quite a while. I couldn't get it to work, but wasn't sure if I shold remove. It had a bunch of legacy stuff in it, so I modernized it. can remove if you think so

Reply:
True :laughing: . I had already rewritten it when I tried to get it working, so I thought I might as well commit my changes so the commented code at least makes sense in the context of the surrounding/actual code for the next dev who attempts to fix it